src/include/OSL/oslconfig.h

```diff
@@ -75,34 +152,55 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
     #endif
 #endif // OSL_DEBUG
 
-#ifndef OSL_INLINE
-    #if OSL_DEBUG
-        #define OSL_INLINE inline
-    #else
-        #if __INTEL_COMPILER >= 1100 || (defined(_WIN32) || defined(_WIN64))
-            #define OSL_INLINE __forceinline
-        #else
-            #define OSL_INLINE inline __attribute__((always_inline))
-        #endif
-    #endif
+
+// OSL_FORCEINLINE is a function attribute that attempts to make the
+// function always inline. On many compilers regular 'inline' is only
+// advisory. Put this attribute before the function return type, just like
+// you would use 'inline'. Note that if OSL_DEBUG is true, it just becomes
+// ordinary inline.
+#if OSL_DEBUG
+#    define OSL_FORCEINLINE inline
+#elif defined(__CUDACC__)
+#    define OSL_FORCEINLINE __inline__
+#elif defined(__GNUC__) || defined(__clang__) || __has_attribute(always_inline)
+#    define OSL_FORCEINLINE inline __attribute__((always_inline))
+#elif defined(_MSC_VER) || defined(__INTEL_COMPILER)
+#    define OSL_FORCEINLINE __forceinline
+#else
+#    define OSL_FORCEINLINE inline
 #endif
 
-#ifndef OSL_NOINLINE
-    #if (defined(_WIN32) || defined(_WIN64))
-        #define OSL_NOINLINE __declspec(noinline)
-    #else
-        #define OSL_NOINLINE __attribute__((noinline))
-    #endif
+
+// OSL_NOINLINE hints to the compiler that the functions should never be
+// inlined.
+#if defined(__GNUC__) || defined(__clang__) || __has_attribute(noinline)
+#    define OSL_NOINLINE __attribute__((noinline))
+#elif defined(_MSC_VER) || defined(__INTEL_COMPILER)
```

Comment:
Don't think the defined(__INTEL_COMPILER) is necessary, its more of a linuyx vs. windows thing.